server/engine/engine.go

```diff
@@ -159,6 +159,8 @@ func (engine *Engine) waitBeforeEnding() {
 		case <-engine.context.Done():
 		}
 	}
+	// Add DeploymentMetric to Database
+	model.EndStatus(engine.database.Instance)
```

Comment:
Instead of making code in model fetch its own status, let's fetch status from engine here and pass it down to `model.EndStatus()`

Reply:
yes, but the `GetLatestExecution` function created here => [https://github.com/ansible/azure-aap-deployment-driver/blob/main/server/engine/engine.go#L178-L187](url) is a bit different from what I've created in my `telemetry.go`, is that fine anyway ?

Reply:
They have different implementation but return the same thing, that's why I'm suggesting to re-use the code, not duplicate it. And since this code is already running in an engine instance, it will be cleaner to do following here in the engine:
1. Create a new function on engine, call it `GetFinalStatus()` and put code from `model.EndStatus` into it and make it return the status. This can be then re-used by the `server/handler/status.go` instead of having another implementation of getting status in the handler.
2. Call the `GetFinalStatus()` function to get status and pass it to `model.SetMetric()`

Reply:
Ok, I think it looks better now.. let me know please.